packages/transpiler/src/transformations/build-super-calls-for-chain.ts

```diff
@@ -0,0 +1,93 @@
+import { getNodeSources, getConstructor, getContract, idModifierInvocation } from '../solc/ast-utils';
+
+import { getInheritanceChain } from '../solc/get-inheritance-chain';
+import { ContractDefinition, ModifierInvocation, Literal } from '../solc/ast-node';
+import { Artifact } from '../solc/artifact';
+
+// builds an __init call with given arguments, for example
+// ERC20DetailedUpgradable.__init(false, "Gold", "GLD", 18)
+function buildSuperCall(args: Literal[], name: string, source: string): string {
+  let superCall = `\n            ${name}Upgradable.__init(false`;
+  if (args && args.length) {
+    superCall += args.reduce((acc, arg) => {
+      const [, , argSource] = getNodeSources(arg, source);
+      return acc + `, ${argSource}`;
+    }, '');
+  }
+  return superCall + ');';
+}
+
+// builds all the __init calls for the parent of a given contract, for example
+// [ '\n            ContextUpgradable.__init(false);' ]
+function buildSuperCalls(
+  node: ContractDefinition,
+  source: string,
+  contracts: string[],
+  mods: ModifierInvocation[],
+  contractsToArtifactsMap: Record<string, Artifact>,
+): (string | never[])[] {
+  const hasInheritance = node.baseContracts.length;
+  if (hasInheritance) {
+    return [
+      ...node.baseContracts
```

Comment:
I think no, because here we are collecting parent's constructor calls. Linearized inheritance calls are preserved with `getInheritanceChain` in `buildSuperCallsForChain`.